src/crypto/hsm/nxp/PersistentStorageOperationalKeystoreHSM.cpp

```diff
@@ -0,0 +1,274 @@
+    0,
+};
+
+uint8_t getEmpytSlotId()
+{
+    uint8_t i = 0;
+    for (auto & mapping : keyidFabIdMapping)
+    {
+        if (mapping.keyId == kKeyId_NotInitialized && mapping.isPending == false)
+        {
+            break;
+        }
+        i++;
+    }
+    return i;
+}
+
+void PersistentStorageOperationalKeystoreHSM::ResetPendingSlot()
+{
+    uint32_t slotId = mPendingKeypair->GetKeyId() - FABRIC_SE05X_KEYID_START;
+    if (slotId < MAX_KEYID_SLOTS_FOR_FABRICS)
+    {
+        keyidFabIdMapping[slotId].keyId       = kKeyId_NotInitialized;
+        keyidFabIdMapping[slotId].fabricIndex = kUndefinedFabricIndex;
+        keyidFabIdMapping[slotId].isPending   = false;
+    }
+}
+
+bool PersistentStorageOperationalKeystoreHSM::HasOpKeypairForFabric(FabricIndex fabricIndex) const
+{
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), false);
+
+    // If there was a pending keypair, then there's really a usable key
+    if (mIsPendingKeypairActive && (fabricIndex == mPendingFabricIndex) && (mPendingKeypair != nullptr))
+    {
+        ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> mPendingKeypair found");
+        return true;
+    }
+
+    for (auto & mapping : keyidFabIdMapping)
+    {
+        if (mapping.fabricIndex == fabricIndex)
+        {
+            ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> stored keyPair found");
+            return true;
+        }
+    }
+
+    ChipLogProgress(Crypto, "SE05x: HasOpKeypairForFabric ==> No key found");
+    return false;
+}
+
+CHIP_ERROR PersistentStorageOperationalKeystoreHSM::NewOpKeypairForFabric(FabricIndex fabricIndex,
+                                                                          MutableByteSpan & outCertificateSigningRequest)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint8_t slotId = getEmpytSlotId();
+
+    VerifyOrReturnError(slotId < MAX_KEYID_SLOTS_FOR_FABRICS, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+
+    ChipLogProgress(Crypto, "SE05x: New Op Keypair for Fabric %02x", fabricIndex);
+
+    // Replace previous pending keypair, if any was previously allocated
+    ResetPendingKey();
+
+    mPendingKeypair = Platform::New<Crypto::P256KeypairHSM>();
+    VerifyOrReturnError(mPendingKeypair != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    // Key id is created as slotid + start offset of ops key id
+    mPendingKeypair->SetKeyId(FABRIC_SE05X_KEYID_START + slotId);
+
+    err = mPendingKeypair->Initialize();
+    VerifyOrReturnError(err == CHIP_NO_ERROR, CHIP_ERROR_NO_MEMORY);
+
+    mPendingFabricIndex = fabricIndex;
+
+    keyidFabIdMapping[slotId].isPending = true;
+
+    size_t csrLength = outCertificateSigningRequest.size();
+    err              = mPendingKeypair->NewCertificateSigningRequest(outCertificateSigningRequest.data(), csrLength);
+    if (err != CHIP_NO_ERROR)
+    {
+        keyidFabIdMapping[slotId].isPending = false;
```

Comment:
Why is this needed?  Doesn't `ResetPendingKey` handle it?

Reply:
Updated.